client/state/sites/products/README.md

```diff
@@ -0,0 +1,26 @@
+Site Products
```

Comment:
Using the [data-layer](https://github.com/Automattic/wp-calypso/tree/master/client/state/data-layer) here would be a nice-to-have. Our current implementation using Redux thunks is sufficient for our use-case, though.